internal/cmd/deploy/deploy.go

```diff
@@ -0,0 +1,187 @@
+package deploy
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/briandowns/spinner"
+	"github.com/google/go-github/github"
+	"github.com/spf13/cobra"
+	"github.com/zeabur/cli/internal/cmdutil"
+	"github.com/zeabur/cli/internal/util"
+)
+
+type Options struct {
+	name string
+}
+
+func NewCmdDeploy(f *cmdutil.Factory) *cobra.Command {
+	opts := &Options{}
+
+	cmd := &cobra.Command{
+		Use:     "deploy",
+		Short:   "Deploy a service",
+		PreRunE: util.NeedProjectContextWhenNonInteractive(f),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runDeploy(f, opts)
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.name, "name", "", "Service name")
+
+	return cmd
+}
+
+func runDeploy(f *cmdutil.Factory, opts *Options) error {
+	if f.Interactive {
+		return runDeployInteractive(f, opts)
+	} else {
+		return runDeployNonInteractive(f, opts)
+	}
+}
+
+func runDeployNonInteractive(f *cmdutil.Factory, opts *Options) error {
+	repoOwner, repoName, err := getRepoInfo()
+	if err != nil {
+		return err
+	}
+
+	repoID, err := getRepoID(repoOwner, repoName)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println(repoID)
+
+	return nil
+}
+
+func runDeployInteractive(f *cmdutil.Factory, opts *Options) error {
+	s := spinner.New(cmdutil.SpinnerCharSet, cmdutil.SpinnerInterval,
+		spinner.WithColor(cmdutil.SpinnerColor),
+		spinner.WithSuffix(" Fetching repository informations..."),
+	)
+	s.Start()
+	repoOwner, repoName, err := getRepoInfo()
+	if err != nil {
+		return err
+	}
+
+	// Use repo name as default service name
+	if opts.name == "" {
+		opts.name = repoName
+	}
+
+	// Get repo ID and branches concurrently
+	repoIDChan := make(chan int)
+	branchesChan := make(chan []string)
+	errChan := make(chan error)
+
+	go func() {
+		repoID, err := getRepoID(repoOwner, repoName)
+		if err != nil {
+			errChan <- err
+			return
+		}
+		repoIDChan <- repoID
+	}()
+
+	go func() {
+		branches, err := getRepoBranches(repoOwner, repoName)
+		if err != nil {
+			errChan <- err
+			return
+		}
+		branchesChan <- branches
+	}()
+
+	repoID := <-repoIDChan
+	branches := <-branchesChan
+
+	// Check for errors
+	select {
+	case err := <-errChan:
+		return err
+	default:
+	}
```

Comment:
Here's the problem which cause deadlock:

1. `repoIDChan` and `branchesChan` are `unbuffered channel`(len is 0). That means, `send` and `receive` action will remain blocked, and only when both operations (send, receive) are initiated **simultaneously** will it proceed with the execution.
2. The sending action of `repoIDChan` and `branchesChan` are in Goroutines. Therefore, it doesn't matter. The behavior is:  **No matter which of the two channels' send signals arrives first, `branchesChan` must wait for the signal from `repoIDChan` to arrive before it can execute.**
3. However, when error occurs, you just `return`. So `Line 100` will be blocked permanently`